docs/contracts.rst

```diff
@@ -1015,6 +1030,8 @@ default constructor: ``contructor() public {}``.
 
 A constructor set as ``internal`` causes the contract to be marked as :ref:`abstract <abstract-contract>`.
 
+A constructor returns the code of the contract, calculating the gas cost depending on the length of the code. The supplied gas may not be enough for this cost, and this is the only situation where an "out of gas" exception does not revert changes to the state, in this case the initialisation of the state variables.
```

Comment:
Hm, I think we need to be a bit more clear here. The constructor is not calculating gas costs itself.

Reply:
@axic Took another try and clarifying, but I am starting to wonder if the note should be somewhere else in the future anyway.